erpnext/stock/get_item_details.py

```diff
@@ -503,7 +503,7 @@ def get_price_list_rate_for(args, item_code):
 	if item_price_data:
 		if item_price_data[0][2] == args.get("uom"):
 			return item_price_data[0][1]
-		elif not args.get('price_list_uom_dependant'):
+		elif args.get('price_list_uom_dependant'):
```

Comment:
Why it is wrong? If price list  is uom dependent, it should only fetch rate if uom matches or no uom mentioned. It should not fetch by multiplying conversion factor.

Reply:
This has caused some confusion so I'll explain it a bit elaborately.

According to the history in ERPNext's GitHub, before `price_not_uom_dependant` field in Price List DocType was introduced, the default behavior was that the price would be calculated after multiplying the conversion factor (I saw it in a GIF). But somehow it got mixed up, such that multi uom tests failed but a workaround was added to make the test pass.

Another way of seeing it: "Price Not UOM Dependant" is the same as "Price is UOM Independent", meaning do not change the price according to conversion factor. So if this field is unchecked then the price should be calculated according to the conversion factor.

Before (bugged):
![uom_dependent_bug](https://user-images.githubusercontent.com/328330/52978166-f01bcf00-33f1-11e9-81a8-a128ae89665e.gif)

After (fixed):
![uom_dependent_fix](https://user-images.githubusercontent.com/328330/52978173-f7db7380-33f1-11e9-9cc8-b2c05973add9.gif)

Reply:
This PR https://github.com/frappe/erpnext/pull/11836 will show that if the Price Not UOM Dependant is checked only then the price should not be multiplied with conversion factor.